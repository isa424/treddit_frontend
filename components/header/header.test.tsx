import React from 'react';
import Header from './header';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { Provider } from 'next-auth/client';

describe('Header component', () => {
	it('should match the snapshot', () => {
		const session = {
			accessToken:
				'3fb32488c440d3c29f7e6d719543786e58ae62083ccf722b5877b5371f9f1a24',
			expires: '2021-03-15T10:53:26.679Z',
			user: {
				image: 'https://avatars.githubusercontent.com/u/5929340?v=4',
			},
		};
		const tree = renderer
			.create(
				<Provider session={session}>
					<Header />
				</Provider>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	afterEach(cleanup);
});
