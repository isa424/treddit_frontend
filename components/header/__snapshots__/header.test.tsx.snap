// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match the snapshot 1`] = `
<nav
  className="header__StyledNav-sc-1cwp5wr-0 kEIQzi navbar navbar-expand-lg navbar-light bg-light fixed-top"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Brand
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li>
          <div
            className="input-group"
          >
            <input
              className="header__Input-sc-1cwp5wr-1 escPUM form-control"
              placeholder="Search..."
              type="text"
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                id="basic-addon2"
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "display": "inline-block",
                      "margin": 0,
                      "maxWidth": "100%",
                      "overflow": "hidden",
                      "position": "relative",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "display": "block",
                        "maxWidth": "100%",
                      }
                    }
                  >
                    <img
                      alt=""
                      aria-hidden={true}
                      role="presentation"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                      style={
                        Object {
                          "border": "none",
                          "display": "block",
                          "margin": 0,
                          "maxWidth": "100%",
                          "padding": 0,
                        }
                      }
                    />
                  </div>
                  <img
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": undefined,
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "visibility": "hidden",
                        "width": 0,
                      }
                    }
                  />
                </div>
              </span>
            </div>
          </div>
        </li>
      </ul>
      <a
        className="header__TallLink-sc-1cwp5wr-3 bmAPwi btn btn-outline-success d-flex align-items-center shadow-sm ml-2"
        href="/posts/create"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 fa-fw "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <i
          className="fas fa-plus"
        />
        <span
          className="ml-2"
        >
          New Post
        </span>
      </a>
      <div
        className="dropdown__Dropdown-sc-1cr2k0u-0 fWspKB dropdown"
      >
        <button
          className="dropdown__Button-sc-1cr2k0u-3 hdErVZ btn btn-light dropdown-toggle p-0 ml-3"
          id="dropdown"
          onClick={[Function]}
          type="button"
        >
          <div
            className="d-flex align-items-center"
          >
            <div
              className="dropdown__Profile-sc-1cr2k0u-1 clnSXD"
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "margin": 0,
                    "maxWidth": "100%",
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "display": "block",
                      "maxWidth": "100%",
                    }
                  }
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style={
                      Object {
                        "border": "none",
                        "display": "block",
                        "margin": 0,
                        "maxWidth": "100%",
                        "padding": 0,
                      }
                    }
                  />
                </div>
                <img
                  className="dropdown__RoundedImage-sc-1cr2k0u-2 hpZcaQ"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
            <span
              className="ml-2"
            />
          </div>
        </button>
        <div
          className="dropdown-menu"
          id="dropdown-menu"
        >
          <a
            className="align-items-center d-flex dropdown-item px-2"
            href="/profile"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-user fa-w-14 fa-fw "
              data-icon="user"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <span
              className="ml-2"
            >
              Profile
            </span>
          </a>
          <a
            className="align-items-center d-flex dropdown-item px-2"
            href="/settings"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-cog fa-w-16 fa-fw "
              data-icon="cog"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <span
              className="ml-2"
            >
              Settings
            </span>
          </a>
          <div
            className="dropdown-divider"
          />
          <button
            className="btn btn-link align-items-center d-flex dropdown-item px-2"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-sign-out-alt fa-w-16 fa-fw "
              data-icon="sign-out-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <span
              className="ml-2"
            >
              Logout
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
